### PR TITLE
JAMES-2741 refactor blobstore save

### PR DESCRIPTION
**Pull request by Arsnael**
Decided to add a save for Strings method in the blob store after discussion with @mbaechler here : https://github.com/linagora/james-project/pull/2310#discussion_r276571030

### CODE REVIEW

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java [hunk 1]
```diff
@@ -26,6 +26,8 @@
 
     Mono<BlobId> save(byte[] data);
 
+    Mono<BlobId> save(String data);
```
Comment (trantienduchn) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java]: it could be declared as a default method

Comment (Arsnael) [reply server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java]: hm I hesitated with that but got the remark once or twice that it was sometimes better to not declare a default method to enforce and make sure we don't forget it in some implementations if the behavior needs to be different. What are the Mokas thinking regarding this?

**Comment by Arsnael** [comment]
```
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] [ERROR] Errors: 
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] [ERROR]   CassandraEventDeadLettersTest » ExceptionInInitializer
```
test this please

**Comment by Arsnael** [comment]
```
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] [ERROR] Failures: 
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] [ERROR]   CassandraMessageIdToImapUidDAOTest.updateShouldUpdateFlaggedFlag:303 
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] Expecting:
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc]   <[ComposedMessageIdWithMetaData{composedMessageId=ComposedMessageId{mailboxId=CassandraId{id=ea74aa71-64ea-11e9-83f0-5b173100cbc4}, messageId=CassandraMessageId{uuid=ea74aa70-64ea-11e9-83f0-5b173100cbc4}, uid=MessageUid{uid=1}}, flags=, modSeq=1}]>
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] to contain only:
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc]   <[ComposedMessageIdWithMetaData{composedMessageId=ComposedMessageId{mailboxId=CassandraId{id=ea74aa71-64ea-11e9-83f0-5b173100cbc4}, messageId=CassandraMessageId{uuid=ea74aa70-64ea-11e9-83f0-5b173100cbc4}, uid=MessageUid{uid=1}}, flags=\Flagged, modSeq=2}]>
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] elements not found:
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc]   <[ComposedMessageIdWithMetaData{composedMessageId=ComposedMessageId{mailboxId=CassandraId{id=ea74aa71-64ea-11e9-83f0-5b173100cbc4}, messageId=CassandraMessageId{uuid=ea74aa70-64ea-11e9-83f0-5b173100cbc4}, uid=MessageUid{uid=1}}, flags=\Flagged, modSeq=2}]>
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] and elements not expected:
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc]   <[ComposedMessageIdWithMetaData{composedMessageId=ComposedMessageId{mailboxId=CassandraId{id=ea74aa71-64ea-11e9-83f0-5b173100cbc4}, messageId=CassandraMessageId{uuid=ea74aa70-64ea-11e9-83f0-5b173100cbc4}, uid=MessageUid{uid=1}}, flags=, modSeq=1}]>
```
test this please

**Comment by Arsnael** [comment]
```
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] # There is insufficient memory for the Java Runtime Environment to continue.
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] 
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] # Native memory allocation (mmap) failed to map 6882852864 bytes for committing reserved memory.
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] 
[4f174758eaff3e4cbf6f0c81c4d9132b160d56bc] OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x0000000625c00000, 6882852864, 0) failed; error='Cannot allocate memory' (errno=12)
```
should we get worried?

**Comment by Arsnael** [comment]
test this please

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/MetricableBlobStore.java [hunk 2]
```diff
@@ -33,6 +33,7 @@
 
     static final String BLOB_STORE_METRIC_PREFIX = "blobStore:";
     static final String SAVE_BYTES_TIMER_NAME = BLOB_STORE_METRIC_PREFIX + "saveBytes";
+    static final String SAVE_STRING_TIMER_NAME = BLOB_STORE_METRIC_PREFIX + "saveString";
```
Comment (trantienduchn) [inline server/blob/blob-api/src/main/java/org/apache/james/blob/api/MetricableBlobStore.java]: If you add one more metric, I'm worry you have to modify the grafana board to add this metric.

File: server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java (hunk 1 above)
Comment (rouazana) [reply server/blob/blob-api/src/main/java/org/apache/james/blob/api/BlobStore.java]: nothing against default methods when appropriate

**Comment by Arsnael** [comment]
```
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR]   WebAdminServerIntegrationTest.postShouldDoMigrationAndUpdateToTheLatestVersion:290 1 expectation failed.
[713d888b07d4495707cfcd6018936885cf9586d1] Response body doesn't match expectation.
[713d888b07d4495707cfcd6018936885cf9586d1] Expected: is "{\"version\":7}"
[713d888b07d4495707cfcd6018936885cf9586d1]   Actual: {"version":6}
```
test this please

**Comment by Arsnael** [comment]
```
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] Error occurred in starting fork, check output in log
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] Process Exit Code: 100
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] Crashed tests:
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] org.apache.james.webadmin.routes.GroupsRoutesTest$ExceptionHandling
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: ExecutionException The forked VM terminated without properly saying goodbye. VM crash or System.exit called
```
test this please

**Comment by Arsnael** [comment]
```
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR] Errors: 
[713d888b07d4495707cfcd6018936885cf9586d1] [ERROR]   RabbitMQMailQueueTest.browseShouldReturnCurrentlyEnqueuedMailFromAllSlices » IllegalState
```
I think the CI hates me today...

test this please

**Comment by Arsnael** [comment]
Was :green_apple: before squashing

**Pull request merged**